pkg/didcomm/common/service/did_comm_msg.go

```diff
@@ -48,6 +49,11 @@ func NewDIDCommMsgMap(payload []byte) (DIDCommMsgMap, error) {
 	return msg, nil
 }
 
+// NewDIDCommMsgMap converts structure(model) to DIDCommMsgMap
+func NewDIDCommMsgMap(v interface{}) DIDCommMsgMap {
+	return toMap(v)
```

Comment:
Can't use mapstructure here as well?

Reply:
I tried it, but unfortunately, it doesn't work as I expected!

Reply:
can you put a note in a comment in there?

Reply:
Done! 
```
	// NOTE: do not try to replace it with mapstructure pkg
	// it doesn't work as expected, at least time.Time won't be converted
```